pkg/globalmanager/downstream.go

```diff
@@ -129,37 +129,82 @@ func (dc *DownstreamController) syncModelWithName(nodeName, modelName, namespace
 
 // syncIncrementalJob syncs the incremental learning jobs
 func (dc *DownstreamController) syncIncrementalJob(eventType watch.EventType, job *sednav1.IncrementalLearningJob) error {
-	// Here only propagate to the nodes with non empty name
+	dataName := job.Spec.Dataset.Name
+	ds, err := dc.client.Datasets(job.Namespace).Get(context.TODO(), dataName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("dataset(%s/%s) not found", job.Namespace, dataName)
+	}
+	dsNodeName := ds.Spec.NodeName
 
-	// FIXME(llhuii): only the case that all workers having the same nodeName are support,
-	// will support Spec.NodeSelector and differenect nodeName.
-	nodeName := job.Spec.TrainSpec.Template.Spec.NodeName
-	if len(nodeName) == 0 {
-		return fmt.Errorf("empty node name")
+	trainNodeName := job.Spec.TrainSpec.Template.Spec.NodeName
+	if len(trainNodeName) == 0 {
+		return fmt.Errorf("train spec: empty node name")
+	}
+
+	evalNodeName := job.Spec.EvalSpec.Template.Spec.NodeName
+	if len(evalNodeName) == 0 {
+		return fmt.Errorf("eval spec: empty node name")
 	}
+
 	dc.injectSecret(job, job.Spec.CredentialName)
 
-	// Sync the model info to edgenode when the job is created
-	if eventType == watch.Added {
-		models := make(map[string]bool)
-		for _, modelName := range []string{
-			job.Spec.InitialModel.Name,
-			job.Spec.DeploySpec.Model.Name,
-		} {
-			models[modelName] = true
+	jobConditions := job.Status.Conditions
+	var latestCondition sednav1.ILJobCondition = sednav1.ILJobCondition{}
+	if len(jobConditions) > 0 {
+		// get latest pod and pod status
+		latestCondition = (jobConditions)[len(jobConditions)-1]
+	} else {
+		return nil
+	}
+
+	currentType := latestCondition.Type
+	jobStage := latestCondition.Stage
+
+	switch currentType {
+	case sednav1.ILJobStageCondWaiting:
+		if err := dc.messageLayer.SendResourceObject(dsNodeName, eventType, job); err != nil {
+			klog.Warningf("Error to sync incremental learning job %s to node %s in stage %s: %v",
+				job.Name, dsNodeName, jobStage, err)
 		}
 
-		for modelName := range models {
-			err := dc.syncModelWithName(nodeName, modelName, job.Namespace)
-			if err != nil {
-				klog.Warningf("Error to sync model %s when sync incremental learning job %s to node %s: %v", modelName, job.Name, nodeName, err)
+		if jobStage == sednav1.ILJobTrain {
+			initModelName := job.Spec.InitialModel.Name
+			if err := dc.syncModelWithName(dsNodeName, initModelName, job.Namespace); err != nil {
+				klog.Warningf("Error to sync model %s when sync incremental learning job %s to node %s: %v",
+					initModelName, job.Name, trainNodeName, err)
+			}
+		}
+
+		if jobStage == sednav1.ILJobEval {
```

Comment:
We can merge this two cases into calling syncModelWithName only once.

Reply:
fix it